CMake Error and Kernel Panic in SiFive FU540

I want to build in fu540. My command is
mkdir build
cd build
cmake  ..  -DLINUX_SIFIVE=y 

But I meet the error 

CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "buildroot" of type UTILITY
    depends on "image-deps" (strong)
  "sm" of type UTILITY
    depends on "linux" (strong)
  "linux" of type UTILITY
    depends on "sysroot" (strong)
  "driver" of type UTILITY
    depends on "linux" (strong)
  "sysroot" of type UTILITY
    depends on "buildroot" (strong)
  "image-deps" of type UTILITY
    depends on "sm" (strong)
    depends on "driver" (strong)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
CMake Generate step failed.  Build files cannot be regenerated correctly.

can you give me some advices

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -244,23 +244,22 @@ else()
   set(enabled_sm --enable-sm)
 endif()
 
+if(firesim)
+  add_patch("riscv-pk" "riscv-pk.firesim.patch" ${sm_srcdir} sm_patches)
+elseif(sifive)
+  add_patch("riscv-pk" "riscv-pk.sifive.patch" ${sm_srcdir} sm_patches)
+endif()
+
 add_custom_command(OUTPUT ${sm_wrkdir}/Makefile WORKING_DIRECTORY ${sm_wrkdir}
   DEPENDS ${sm_wrkdir_exists}
   COMMAND ${sm_srcdir}/configure --host=riscv${BITS}-unknown-linux-gnu --with-payload=${linux_vmlinux_stripped}
     --enable-logo --with-logo=${confdir}/sifive_logo.txt ${enabled_sm} ${SM_CONFIGURE_ARGS} --with-target-platform=${platform}
   COMMENT "Configuring sm"
 )
-add_custom_target("sm" ALL DEPENDS ${sm_wrkdir}/Makefile "linux" WORKING_DIRECTORY ${sm_wrkdir}
+add_custom_target("sm" ALL DEPENDS ${sm_wrkdir}/Makefile "linux" ${sm_patches} WORKING_DIRECTORY ${sm_wrkdir}
   COMMAND env CFLAGS='${CFLAGS} -mabi=${ABI} -march=${ISA}' $(MAKE) -C ${sm_wrkdir}
   COMMENT "Building sm"
 )
-
-if(firesim)
-  add_patch("riscv-pk" "riscv-pk.firesim.patch" ${sm_srcdir} "sm")
-elseif(sifive)
-  add_patch("riscv-pk" "riscv-pk.sifive.patch" ${sm_srcdir} "sm")
-endif()
-
 
 ###############################################################################
 ## COMPONENT: tests

